src/Shared/DomainEventWrapper.cs

```diff
@@ -10,25 +11,46 @@ public DomainEventWrapper(EventGridEvent eventGridEvent)
     {
         this.Data = eventGridEvent.Data.ToObjectFromJson<JsonObject>();
         this.DomainEventName = eventGridEvent.EventType;
+        this.DomainEventSchema = EventSchema.EventGridEvent;
     }
 
-    private DomainEventWrapper(JsonObject data, string domainEventName)
+    public DomainEventWrapper(CloudEvent cloudEvent)
+    {
+        this.Data = cloudEvent.Data!.ToObjectFromJson<JsonObject>();
+        this.DomainEventName = cloudEvent.Type;
+        this.DomainEventSchema = EventSchema.CloudEvent;
+    }
+
+    private DomainEventWrapper(JsonObject data, string domainEventName, EventSchema schema)
     {
         this.Data = data;
         this.DomainEventName = domainEventName;
+        this.DomainEventSchema = schema;
     }
 
     public JsonObject Data { get; }
 
     public string DomainEventName { get; }
 
+    public EventSchema DomainEventSchema { get; }
+
     public void SetMetadata(string key, string value)
     {
+        if (this.DomainEventSchema == EventSchema.CloudEvent)
```

Comment:
Do we need this given open telemetry has not been implemented for pull model? Worst case if the behavior is registered by mistake you'll get tracing for free maybe? Probably not how it's intended for CloundEvents but at least it won't break

Reply:
As we discussed yesterday, we will keep _as-is_ for the time being. It can always be removed when cloud events are fully supported by the library.